### DISCUSSION TITLE
HomeKit I need help!

### DISCUSSION THREAD

**Discussion by sotidis** [HomeSpan/HomeSpan#79] [Q&A]
DEVICE NOT YET PAIRED -- PLEASE PAIR WITH HOMEKIT APP

** Client #0 Connected: (25 sec) 192.168.2.57 on Socket 2/10
In Pair Setup...In Pair Setup...In Pair Setup...
*** ACCESSORY PAIRED! ***
** Client #0 Connected: (36 sec) 192.168.2.57 on Socket 3/10
In Get Accessories #0 (192.168.2.57)...
In Post Pairings #0 (192.168.2.57)...Remove...
That was last Admin Controller!  Removing any remaining Regular Controllers and unpairing Accessory
*** Terminating Client #0
** Disconnecting Client #0  (36 sec)

**Comment by HomeSpan** [comment]
Can you please provide more details.  What chip are you using, how are you starting the pairing function, etc.  Also, please turn on level 2 logging to show all the diagnostic messages.

**Comment by sotidis** [comment]
ESP WROOM 32 (tried two different chips) and RFM69 FeatherWing from Adafruit and iPad Air (12.5.2)

*** Log Level set to 2

16:59:43.160 -> =======================================
16:59:43.160 -> ** Client #0 Connected: (246 sec) 192.168.2.57 on Socket 2/10
16:59:43.160 -> 
16:59:43.160 -> <<<<<<<<< 192.168.2.57 <<<<<<<<<
16:59:43.203 -> POST /pair-setup HTTP/1.1
16:59:43.203 -> Host: Somfy-HomeSpan._hap._tcp.local
16:59:43.203 -> Content-Length: 6
16:59:43.203 -> Content-Type: application/pairing+tlv8
16:59:43.203 -> ------------ END BODY! ------------
16:59:43.203 -> STATE(1) 01
16:59:43.203 -> METHOD(1) 00
16:59:43.203 -> ------------ END TLVS! ------------
16:59:43.203 -> In Pair Setup...Found <M1>.  Expected <M1>
16:59:43.903 -> 
16:59:43.903 -> >>>>>>>>>> 192.168.2.57 >>>>>>>>>>
16:59:43.903 -> HTTP/1.1 200 OK
16:59:43.903 -> Content-Type: application/pairing+tlv8
16:59:43.903 -> Content-Length: 409
16:59:43.903 -> 
16:59:43.903 -> STATE(1) 02
16:59:43.903 -> PUBKEY(384) AEADE14ADD89B5933AEC0441634ADB49FC5AECE48D640C7F8FE8E08622103CD99BF4C4C32713DFA982324088EED0F3619C21D40C6D18272AC0433EFEFD9FCC66D13D660411DB5D1C9A87262746627296B0EDEF152EEE145E35D81F4FB870FF987E170B3FD6D6EEEE5D38663CBBCAE635AF156C60BA28209A96104C3F191B1E844C94E0DDFE8D39DE217B42C2742459913F1007371A04C0675E526D24DEAFAC67484E6EDB818237A749B6A0C78A6C15F4DBEA48DBCAEACD43F1749633734B91357D8076B305C3793AE5B8272401BF6A309C09CC3781E3CAFA6D508ADAADC86FDA4DB975890A635A3AD3CFF970BE6A98A32DB06EC1B5144E60293D787B30B59D7FEAABAF94D4D33757727E50B50837ACA10C87EE58B511094AC9246A8294191BA87A578BA27B4781A1369ACE212DD43371954FF02CAEB536764A8B8388AB31A78F1D6E8DFDFF4F598F22A84675529122D7EDBE57825DDF5C9D373ABA26831BB116DD1CC9E74DD86D8BBAE91E762FB80873D66F35E806294CDAD3001664F7248D2E
16:59:43.981 -> SALT(16) 4A42D4E6C34E9912D8B6B7FEA427EADC
16:59:43.981 -> ------------ SENT! --------------
16:59:43.981 -> 
16:59:51.504 -> <<<<<<<<< 192.168.2.57 <<<<<<<<<
16:59:51.504 -> POST /pair-setup HTTP/1.1
16:59:51.545 -> Host: Somfy-HomeSpan._hap._tcp.local
16:59:51.545 -> Content-Length: 457
16:59:51.545 -> Content-Type: application/pairing+tlv8
16:59:51.545 -> ------------ END BODY! ------------
16:59:51.545 -> STATE(1) 03
16:59:51.545 -> PUBKEY(384) E9C07D0E1119D8CD0CB8E6BA319C03B07A16A559EE76D0E5772D29628DDCE2A47A6DE50D189EE7F1FC0D1DE4417870B342F40058067933042B33EE45723229F170F295EAB9D310AFA33E6677693E425ADD3C3C456BCCDAE04EB840FA1B40FFBFED5AC2787F317A2785AB7E86E96DD456C798D14B7179C7C46EC9A102FCABADE0CA45FD7BA23E19B5F853F6313B73712271924AA46CDE79DAE4735014BB9D66E2D92D6F4648FD66DEC774FDC700EF1BA53EEDBDDE66FD44554382C1C8861C29C2A4197DCE62E43C50999F072ADC40DEBA99787A08951E593527E1A588ABC2D26637166339F2FB1FC279B70D696C90F4F54AC01BF1385000299ADADE509618326A95C56898D77EDBA8DF90D3E91F664F2728565810EF69069E9D89E4A423785034F2A8C3B2B0E335AD428421DA9A0D222B9E0B2D9B22C22A51A184B1FD685DAE77E6A9758480D7C553E95EC0B303B025BEE591DDF274A22FE0CE70875E687DCEB23FA06384EB2B94F9F6C2B6E5C2154BC94751DFB06E75661A36523178BB19C60C
16:59:51.606 -> PROOF(64) 76A33AD04A749C4F37413B386AFB5D406D447AFFACDA87B6CBA0BB887A4370B4932A8E41E450A4B32F5084AA5E1A1C1F6FAF0EAC7CA5C48CC1DC7F756319E85A
16:59:51.606 -> ------------ END TLVS! ------------
16:59:51.606 -> In Pair Setup...Found <M3>.  Expected <M3>
16:59:53.711 -> 
16:59:53.711 -> >>>>>>>>>> 192.168.2.57 >>>>>>>>>>
16:59:53.711 -> HTTP/1.1 200 OK
16:59:53.711 -> Content-Type: application/pairing+tlv8
16:59:53.711 -> Content-Length: 69
16:59:53.753 -> 
16:59:53.753 -> STATE(1) 04
16:59:53.753 -> PROOF(64) 45EAE79D587F28F4C6844CD97B4242A6217AEDC80A72E52130BD493ADFB1C2D03C60CB26252E5A06AD99A74521E926B000E54DAEF982EF117A43BEDD29BA28AB
16:59:53.753 -> ------------ SENT! --------------
16:59:53.753 -> 
16:59:55.410 -> <<<<<<<<< 192.168.2.57 <<<<<<<<<
16:59:55.410 -> POST /pair-setup HTTP/1.1
16:59:55.410 -> Host: Somfy-HomeSpan._hap._tcp.local
16:59:55.410 -> Content-Length: 159
16:59:55.410 -> Content-Type: application/pairing+tlv8
16:59:55.410 -> ------------ END BODY! ------------
16:59:55.410 -> STATE(1) 05
16:59:55.444 -> ENC.DATA(154) 55A7260B343DD60E53AE52B4D5BEC137255E8292B6E0BCC1A35694E35DA922219DA99A0651739432A99364457F09DDA7978C1F9023EF8273DC584863B812059529A2840726E301BE2AF503991270A1C0D59C68486E93AA64D8C8F0157345B12E04CF0EF49B22014590D4D1FC800179C5F4F686E77E992F9475C70112AF7687E32DC2B1D7E8B19779A0EC11D5CC2E21FABA6F3A987D5C84FB3CD7
16:59:55.444 -> ------------ END TLVS! ------------
16:59:55.444 -> In Pair Setup...Found <M5>.  Expected <M5>
16:59:55.444 -> PUBKEY(32) 43B8405906C2698F0802EDD8A5C29C822873AAA2EF88B3550E68C0A0EDDA10FC
16:59:55.477 -> SIGNATURE(64) AB982DD9D7DA5005E86D119AC6FA796B0E717E7A3FB20830F5C173DA6F2D25B0D2CA94698CDBE7985CD64A36543B11F1ED27D0079BC7AA18A628733B753E6902
16:59:55.477 -> IDENTIFIER(36) 42333835374542352D333338322D344331372D423041442D384245363245343739323735
16:59:55.477 -> ------- END DECRYPTED TLVS! -------
16:59:55.544 -> 
16:59:55.544 -> *** Added Controller: B3857EB5-3382-4C17-B0AD-8BE62E479275 (admin)
16:59:55.544 -> 
16:59:55.578 -> ------- ENCRYPTING SUB-TLVS -------
16:59:55.578 -> PUBKEY(32) 9EA176160FEFFCF764D4E49DD72F76536097405D2D15280C3187D1343C06395D
16:59:55.578 -> SIGNATURE(64) 186C2EF7C4FB93D8A751539E6B17B22DAB9AE9E027946161B0345724292FE68C43361372E5F5F25D65D69F1B5252EB584D193C67BED0CDA45EDB82937E942C02
16:59:55.612 -> IDENTIFIER(17) 46363A34393A33363A37363A45433A3746
16:59:55.612 -> ---------- END SUB-TLVS! ----------
16:59:55.612 -> 
16:59:55.612 -> >>>>>>>>>> 192.168.2.57 >>>>>>>>>>
16:59:55.612 -> HTTP/1.1 200 OK
16:59:55.612 -> Content-Type: application/pairing+tlv8
16:59:55.612 -> Content-Length: 140
16:59:55.612 -> 
16:59:55.612 -> STATE(1) 06
16:59:55.612 -> ENC.DATA(135) 4217554870BDA4908EEAC404BABFAEC11A36F5588E763835000FD0928E264F4E52D5235B087171BDBA885ED332BD85A9E268FD94E31D88FD7394B11164C5DA2E90CD30BF4FFBB644D9F471B59976A8DDB7F3639A247A22262288761F468078B49EB7386206A4C11B9C4C5F8843300F458C4D1A5E2CF1819040A7820A372DAB2492591B1BB01A57
16:59:55.645 -> ------------ SENT! --------------
16:59:55.645 -> 
16:59:55.645 -> *** ACCESSORY PAIRED! ***
16:59:55.645 -> 
16:59:57.069 -> =======================================
16:59:57.069 -> ** Client #0 Connected: (260 sec) 192.168.2.57 on Socket 3/10
16:59:57.069 -> 
16:59:57.069 -> <<<<<<<<< 192.168.2.57 <<<<<<<<<
16:59:57.069 -> POST /pair-verify HTTP/1.1
16:59:57.069 -> Host: Somfy-HomeSpan._hap._tcp.local
16:59:57.069 -> Content-Length: 37
16:59:57.069 -> Content-Type: application/pairing+tlv8
16:59:57.069 -> ------------ END BODY! ------------
16:59:57.069 -> STATE(1) 01
16:59:57.069 -> PUBKEY(32) E33ED8DFB2460A09E85F10BA9B48D7CDD4F918818037A67BF3290D8128833B11
16:59:57.115 -> ------------ END TLVS! ------------
16:59:57.115 -> In Pair Verify #0 (192.168.2.57)...Found <M1>
16:59:57.173 -> ------- ENCRYPTING SUB-TLVS -------
16:59:57.173 -> SIGNATURE(64) A597DCAC17F1040A9521FB70B73890D6CC4F7F854ED5E67F27CDAF74B4BBA5AA4BDED22174FB30A7C9E310C52E9477C848CE236C7797BDC71D9BA82BCBDE7308
16:59:57.207 -> IDENTIFIER(17) 46363A34393A33363A37363A45433A3746
16:59:57.207 -> ---------- END SUB-TLVS! ----------
16:59:57.207 -> 
16:59:57.207 -> >>>>>>>>>> 192.168.2.57 >>>>>>>>>>
16:59:57.207 -> HTTP/1.1 200 OK
16:59:57.207 -> Content-Type: application/pairing+tlv8
16:59:57.207 -> Content-Length: 140
16:59:57.207 -> 
16:59:57.207 -> STATE(1) 02
16:59:57.207 -> PUBKEY(32) F6B1257B4F070BC6F38240126B874C8F585AB6DBDF35BB598C49F5455C3C683C
16:59:57.207 -> ENC.DATA(101) 715D9C863F519B1F1C12F82F5CFAB7CC96B557313DAE3688A4174507ABF68955432B033BE9261DC3592673F1A971F0D90A35DC6C56FBABE4544A24C09503EFCD5478D1FDD4FAD649B1F62235B589EE35D3F586A02F7F2701EEB8021A52D996B6093FE6E393
16:59:57.241 -> ------------ SENT! --------------
16:59:57.241 -> 
17:00:00.522 -> <<<<<<<<< 192.168.2.57 <<<<<<<<<
17:00:00.522 -> POST /pair-verify HTTP/1.1
17:00:00.522 -> Host: Somfy-HomeSpan._hap._tcp.local
17:00:00.522 -> Content-Length: 125
17:00:00.522 -> Content-Type: application/pairing+tlv8
17:00:00.559 -> ------------ END BODY! ------------
17:00:00.559 -> STATE(1) 03
17:00:00.559 -> ENC.DATA(120) 4B66684CA6FEEE286385408EFB869F07423A366607D918E05CDE626DDA610FDE1C985A7A04AEA02DFD1D3039F0E9B6646D6F5BCEEB32E1CCB66A8AFA68879F80A005D10792A70629709F1DC2DC62AE251EB206601B9768070D77046CCC6A2FD32BF3A4CCBE885B41C7716C7A1308F9C6475010983A029541
17:00:00.559 -> ------------ END TLVS! ------------
17:00:00.559 -> In Pair Verify #0 (192.168.2.57)...Found <M3>
17:00:00.559 -> SIGNATURE(64) 9A67F4DC78C38568444CF054BA8DF63B2B1E09B659A5794DF53E3362D0CD50564DC8241D629E0FE1DD3B7B62630B649179CF2365CABF8883185C2F1784EC7802
17:00:00.590 -> IDENTIFIER(36) 42333835374542352D333338322D344331372D423041442D384245363245343739323735
17:00:00.590 -> ------- END DECRYPTED TLVS! -------
17:00:00.590 -> Found Controller: B3857EB5-3382-4C17-B0AD-8BE62E479275 (admin)
17:00:00.659 -> 
17:00:00.659 -> >>>>>>>>>> 192.168.2.57 >>>>>>>>>>
17:00:00.659 -> HTTP/1.1 200 OK
17:00:00.659 -> Content-Type: application/pairing+tlv8
17:00:00.659 -> Content-Length: 3
17:00:00.659 -> 
17:00:00.659 -> STATE(1) 04
17:00:00.659 -> ------------ SENT! --------------
17:00:00.659 -> 
17:00:00.659 -> *** SESSION VERIFICATION COMPLETE *** 
17:00:00.659 -> 
17:00:01.134 -> <<<< #### 192.168.2.57 #### <<<<
17:00:01.134 -> GET /accessories HTTP/1.1
17:00:01.134 -> Host: Somfy-HomeSpan._hap._tcp.local
17:00:01.172 -> ------------ END BODY! ------------
17:00:01.172 -> In Get Accessories #0 (192.168.2.57)...
17:00:01.172 -> 
17:00:01.172 -> >>>>>>>>>> 192.168.2.57 >>>>>>>>>>
17:00:01.172 -> HTTP/1.1 200 OK
17:00:01.172 -> Content-Type: application/hap+json
17:00:01.172 -> Content-Length: 1375
17:00:01.172 -> 
17:00:01.172 -> {"accessories":[{"aid":1,"services":[{"iid":1,"type":"3E","characteristics":[{"iid":2,"type":"23","value":"Somfy Controller","format":"string","perms":["pr"]},{"iid":3,"type":"20","value":"HomeSpan","format":"string","perms":["pr"]},{"iid":4,"type":"30","value":"123-ABC","format":"string","perms":["pr"]},{"iid":5,"type":"21","value":"Multi-Channel RTS","format":"string","perms":["pr"]},{"iid":6,"type":"52","value":"1.0.1","format":"string","perms":["pr"]},{"iid":7,"type":"14","format":"bool","perms":["pw"]}]},{"iid":8,"type":"A2","characteristics":[{"iid":9,"type":"37","value":"1.1.0","format":"string","perms":["pr"]}]}]},{"aid":3,"services":[{"iid":1,"type":"3E","characteristics":[{"iid":2,"type":"23","value":"Channel-2","format":"string","perms":["pr"]},{"iid":3,"type":"20","value":"HomeSpan","format":"string","perms":["pr"]},{"iid":4,"type":"30","value":"RTS-82D368","format":"string","perms":["pr"]},{"iid":5,"type":"21","value":"Channel-2","format":"string","perms":["pr"]},{"iid":6,"type":"52","value":"1.0.1","format":"string","perms":["pr"]},{"iid":7,"type":"14","format":"bool","perms":["pw"]}]},{"iid":8,"type":"8C","characteristics":[{"iid":9,"type":"6D","value":0,"format":"uint8","perms":["pr","ev"]},{"iid":10,"type":"7C","value":0,"format":"uint8","perms":["pr","pw","ev"]},{"iid":11,"type":"24","value":0,"format":"bool","perms":["pr","ev"]}]}]}]}
17:00:01.303 -> -------- SENT ENCRYPTED! --------
17:00:01.303 -> 
17:00:01.575 -> <<<< #### 192.168.2.57 #### <<<<
17:00:01.575 -> POST /pairings HTTP/1.1
17:00:01.608 -> Host: Somfy-HomeSpan._hap._tcp.local
17:00:01.608 -> Content-Length: 44
17:00:01.608 -> Content-Type: application/pairing+tlv8
17:00:01.608 -> ------------ END BODY! ------------
17:00:01.608 -> STATE(1) 01
17:00:01.608 -> METHOD(1) 04
17:00:01.608 -> IDENTIFIER(36) 42333835374542352D333338322D344331372D423041442D384245363245343739323735
17:00:01.608 -> ------------ END TLVS! ------------
17:00:01.608 -> In Post Pairings #0 (192.168.2.57)...Remove...
17:00:01.608 -> Found Controller: B3857EB5-3382-4C17-B0AD-8BE62E479275 (admin)
17:00:01.642 -> 
17:00:01.642 -> ***Removed Controller: B3857EB5-3382-4C17-B0AD-8BE62E479275 (admin)
17:00:01.642 -> That was last Admin Controller!  Removing any remaining Regular Controllers and unpairing Accessory
17:00:01.642 -> 
17:00:01.642 -> 
17:00:01.642 -> >>>>>>>>>> 192.168.2.57 >>>>>>>>>>
17:00:01.642 -> HTTP/1.1 200 OK
17:00:01.642 -> Content-Type: application/pairing+tlv8
17:00:01.642 -> Content-Length: 3
17:00:01.642 -> 
17:00:01.642 -> STATE(1) 02
17:00:01.642 -> -------- SENT ENCRYPTED! --------
17:00:01.642 -> *** Terminating Client #0
17:00:01.642 -> ** Disconnecting Client #0  (265 sec)
17:00:01.677 ->

**Comment by sotidis** [comment]
17:20:43.911 -> rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
17:20:43.946 -> configsip: 0, SPIWP:0xee
17:20:43.946 -> clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
17:20:43.946 -> mode:DIO, clock div:2
17:20:43.946 -> load:0x3fff0018,len:4
17:20:43.946 -> load:0x3fff001c,len:1216
17:20:43.946 -> ho 0 tail 12 room 4
17:20:43.946 -> load:0x40078000,len:10944
17:20:43.946 -> load:0x40080400,len:6360
17:20:43.946 -> entry 0x400806b4
17:20:46.293 -> 
17:20:46.293 -> ************************************************************
17:20:46.293 -> Welcome to HomeSpan!
17:20:46.328 -> Apple HomeKit for the Espressif ESP-32 WROOM and Arduino IDE
17:20:46.328 -> ************************************************************
17:20:46.328 -> 
17:20:46.328 -> ** Please ensure serial monitor is set to transmit <newlines>
17:20:46.328 -> 
17:20:46.328 -> Message Logs:     Level 1
17:20:46.328 -> Status LED:       Pin 13
17:20:46.328 -> Device Control:   Pin 21
17:20:46.328 -> Sketch Version:   n/a
17:20:46.328 -> HomeSpan Version: 1.2.1
17:20:46.328 -> ESP-IDF Version:  v3.3.5-1-g85c43024c
17:20:46.328 -> ESP32 Board:      esp32
17:20:46.328 -> Sketch Compiled:  Apr 24 2021 16:13:02
17:20:46.362 -> 
17:20:46.362 -> Device Name:      Somfy-HomeSpan
17:20:46.362 -> 
17:20:46.568 -> Found SPI Device with Chip Select Pin=33
17:20:46.568 -> 
17:20:46.603 -> Configuring Somfy Window Shade Channel-2 (RTS-82D368):  RollingCode=00FF  RaiseTime=11000 ms  LowerTime=10000 ms
17:20:46.603 -> 
17:20:46.603 -> *** HomeSpan Info ***
17:20:46.603 -> 
17:20:46.603 -> ➤ Accessory:  AID=1
17:20:46.603 ->    ➟ Service AccessoryInformation:  IID=1, UUID=0x3E
17:20:46.603 ->       ⇨ Characteristic Name("Somfy Controller"):  IID=2, UUID=0x23
17:20:46.603 ->       ⇨ Characteristic Manufacturer("HomeSpan"):  IID=3, UUID=0x20
17:20:46.603 ->       ⇨ Characteristic SerialNumber("123-ABC"):  IID=4, UUID=0x30
17:20:46.637 ->       ⇨ Characteristic Model("Multi-Channel RTS"):  IID=5, UUID=0x21
17:20:46.637 ->       ⇨ Characteristic FirmwareRevision("1.0.1"):  IID=6, UUID=0x52
17:20:46.637 ->       ⇨ Characteristic Identify(0):  IID=7, UUID=0x14
17:20:46.637 ->    ➟ Service HAPProtocolInformation:  IID=8, UUID=0xA2
17:20:46.637 ->       ⇨ Characteristic Version("1.1.0"):  IID=9, UUID=0x37
17:20:46.637 -> ➤ Accessory:  AID=3
17:20:46.637 ->    ➟ Service AccessoryInformation:  IID=1, UUID=0x3E
17:20:46.671 ->       ⇨ Characteristic Name("Channel-2"):  IID=2, UUID=0x23
17:20:46.671 ->       ⇨ Characteristic Manufacturer("HomeSpan"):  IID=3, UUID=0x20
17:20:46.671 ->       ⇨ Characteristic SerialNumber("RTS-82D368"):  IID=4, UUID=0x30
17:20:46.671 ->       ⇨ Characteristic Model("Channel-2"):  IID=5, UUID=0x21
17:20:46.671 ->       ⇨ Characteristic FirmwareRevision("1.0.1"):  IID=6, UUID=0x52
17:20:46.671 ->       ⇨ Characteristic Identify(0):  IID=7, UUID=0x14
17:20:46.705 ->    ➟ Service WindowCovering:  IID=8, UUID=0x8C
17:20:46.705 ->       ⇨ Characteristic CurrentPosition(0):  IID=9, UUID=0x6D  Range=[0,100]
17:20:46.705 ->       ⇨ Characteristic TargetPosition(0):  IID=10, UUID=0x7C  Range=[0,100]
17:20:46.705 ->       ⇨ Characteristic ObstructionDetected(0):  IID=11, UUID=0x24
17:20:46.705 ->       ✘ Characteristic PositionState *** WARNING!  Required Characteristic for this Service not found. ***
17:20:46.738 -> 
17:20:46.738 -> *** CAUTION: There is 1 WARNING associated with this configuration that may lead to the device becoming non-responsive, or operating in an unexpected manner. ***
17:20:46.738 -> 
17:20:46.738 -> Configured as Bridge: YES
17:20:46.738 -> 
17:20:46.738 -> Service                         UUID         AID  IID  Update  Loop  Button  Linked Services
17:20:46.738 -> ------------------------------  ----  ----------  ---  ------  ----  ------  ---------------
17:20:46.771 -> AccessoryInformation              3E           1    1     YES    NO      NO  -
17:20:46.771 -> HAPProtocolInformation            A2           1    8      NO    NO      NO  -
17:20:46.771 -> AccessoryInformation              3E           3    1     YES    NO      NO  -
17:20:46.771 -> WindowCovering                    8C           3    8     YES   YES      NO  -
17:20:46.771 -> 
17:20:46.771 -> *** End Info ***
17:20:46.771 -> 
17:20:46.771 -> Accessory ID:      F6:49:36:76:EC:7F                               LTPK: 9EA176160FEFFCF764D4E49DD72F76536097405D2D15280C3187D1343C06395D
17:20:46.806 -> No Paired Controllers
17:20:46.806 -> 
17:20:46.806 -> Accessory configuration number: 1
17:20:46.806 -> 
17:20:46.806 -> Somfy-HomeSpan is READY!
17:20:46.806 -> 
17:20:46.806 -> Trying to connect to D21.  Waiting 1 second(s) for response...
17:20:47.925 -> Trying to connect to D21.  Waiting 2 second(s) for response...
17:20:49.905 -> Trying to connect to D21.  Waiting 4 second(s) for response...
17:20:52.009 -> Successfully connected to D21! IP Address: 192.168.2.68
17:20:52.009 -> 
17:20:52.009 -> Starting MDNS...
17:20:52.009 -> 
17:20:52.009 -> HostName:      HomeSpan-F6493676EC7F.local:80
17:20:52.009 -> Display Name:  Somfy-HomeSpan
17:20:52.009 -> Model Name:    HomeSpan-ESP32
17:20:52.009 -> Setup ID:      HSPN
17:20:52.009 -> 
17:20:52.009 -> Starting Web (HTTP) Server supporting up to 8 simultaneous connections...
17:20:52.058 -> 
17:20:52.058 -> DEVICE NOT YET PAIRED -- PLEASE PAIR WITH HOMEKIT APP
17:20:52.058 ->

**Comment by HomeSpan** [comment]
Let's start with a simpler set-up to debug - the SOMFY controller is complex and the RFM69 adds additional complexity.  Remove the RFM69 so you are just using the ESP32 chip.  Then load HomeSpan Example 1 and let's see what happens.

Also, do you have any devices with IOS 13 or IOS 14.  There have been significant changes to Apple's HomeKit since IOS12 - want to rule that out as a potential problem.

**Reply by sotidis** [reply]
HomeSpan Example 1-OK! works

(I do not have devices with IOS 13 or IOS 14)

**Comment by HomeSpan** [comment]
My guess is that the issue is related to changes from iOS12 to iOS14.  To check, add the following:

```C++
new Characteristic::PositionState(0);
```

to line 64 of Somfy_DEV.h right after the lines 62 and 63 which should read:

```C++
current=new Characteristic::CurrentPosition(0);    // Windows Shades have positions that range from 0 (fully lowered) to 100 (fully raised)    
    target=new Characteristic::TargetPosition(0);      // Windows Shades have positions that range from 0 (fully lowered) to 100 (fully raised)
```

The PositionState Characteristic is no longer used by HomeKit since iOS13, but iOS12 may still require it.   Based on your log files, right after pairing, your Home App reads the Accessory database and decides it want to unpair from HomeSpan.  The only reason I can think of is that the Home App does not like the configuration, perhaps because it needs the PositionState Characteristic.

**Reply by sotidis** [reply]
Thank you! it works!

**Comment by HomeSpan** [comment]
Excellent. However, I'm not sure if the Somfy sketch itself will work under iOS12.  You may need to make use of the PositionState Characteristic to ensure the Home App responds properly - the HAP R2 guide describes how Position State should be used.  Unfortunately I don't have an iOS12 device so can't design/test an iOS12-specific version of the sketch.  Please let me know how it goes.

**Reply by sotidis** [reply]
2.Press and HOLD the PROG/SELECTOR button on the HomeSpan Somfy device until the window shade or screen once again jogs up and down.

I stopped here , does not connect (((

** Somfy Channel-1: PROGRAMMING  RC=0129
Transmitting: A0 8A 01 29 07 01 26
Obfuscated:   A0 2A 2B 02 05 04 22
** Somfy Channel-1: PROGRAMMING  RC=012A
Transmitting: A0 89 01 2A 07 01 26
Obfuscated:   A0 29 28 02 05 04 22
** Somfy Channel-1: PROGRAMMING  RC=012B
Transmitting: A0 88 01 2B 07 01 26
Obfuscated:   A0 28 29 02 05 04 22
** Somfy Channel-1: PROGRAMMING  RC=012C
Transmitting: A0 8F 01 2C 07 01 26
Obfuscated:   A0 2F 2E 02 05 04 22

How can I check if the RFM69 is working?
